### ISSUE TITLE
CheckBox: a11y issues

### ISSUE THREAD

**Issue by 2bno1** [SAP/ui5-webcomponents#3433]
### **Bug Description**
axe devtool reports 2 issues for CheckBox component.
1. Nested interactive controls are not announced by screen readers
2. Flags elements whose role is none or presentation and which cause the role conflict resolution to trigger.

### **Expected Behavior**
axe devtool should not report a11y issues for CheckBox component.

### **Steps to Reproduce**
1. Install axe devtool (https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd)
2. Go to https://scsxe.csb.app/
3. Run automatic scan using axe devtool

### **Isolated Example**
https://codesandbox.io/s/cranky-snow-scsxe?file=/src/App.js

### **Context**
- UI5 Web Components version: 0.31.6
- OS/Platform: macOS
- Browser: Chrome
- Affected component: CheckBox

### **Log Output / Stack Trace / Screenshots**
<img width="2559" alt="WeChatd9a15686a8c03733b7ce7191765f2f56" src="https://user-images.githubusercontent.com/1751846/122172791-b705a380-ceb3-11eb-9e5c-e53e8027d106.png">
<img width="2559" alt="WeChat60feb059b1e3fe50b242b4d6fe51aea7" src="https://user-images.githubusercontent.com/1751846/122172938-d56b9f00-ceb3-11eb-9b4f-4a1513d769cb.png">
### **Priority**
- [ ] Low
- [x] Medium
- [ ] High
- [ ] Very High

**Comment by gmkv** [comment]
Hi @SAP/ui5-webcomponents-topic-p 

There are accessibility cornerns as reported regarding the CheckBox component. 
Could you take a look?
- [Isolated sample with React](https://codesandbox.io/s/cranky-snow-scsxe?file=/src/App.js)
- [Isolated sample in pure HTML](https://codesandbox.io/s/ui5-webcomponents-forked-xk1og?file=/index.html)

Regards,
Georgi

**Comment by dobrinyonkov** [comment]
Hello @2bno1, 

we have merged #3511, which resolves one of the warnings:
- Flags elements whose role is none or presentation and which cause the role conflict resolution to trigger.

As for the first warning: 
- Nested interactive controls are not announced by screen readers

None of the elements inside the checkbox are actually focusable.  We have a hidden input element which is to be used 
if the checkbox component will be submitted in an HTML form. But it is hidden from accessibility tree mapping by aria-hidden="true" and not focusable by tabindex="-1".

Plain HTML example - issue still reported: https://codesandbox.io/s/happy-mestorf-2q51y?file=/index.html
I suggest contacting the AXE community for anwsers. 

Thanks and Best Regards,
Dobrin
